mod/file/actions/file/upload.php

```diff
@@ -68,6 +68,10 @@
 		// user blanked title, but we need one
 		$title = $file->title;
 	}
+
+	if ($entity->access_id !== $access_id) {
```

Comment:
The changes in this file look like a bugfix that could be moved to a separate PR for 2.0, or am I missing something? Also you'd want to set `$reset_icon_urls` either way so you don't get notices below.